### DISCUSSION TITLE
New Widget ComboboxScroll

### DISCUSSION THREAD

**Discussion by develcooking** [slint-ui/slint#5304] [Ideas] [answered]
Hi, I'm currently working on an application that uses a ComboBox like this. I wanted to share my code with upstream.

I had many issues getting started on it. I used #2657 as my starting point. After encountering transparency issues, I found out that the default palette of Fluent has a lot of transparency, so I created some extra brushes.

A big issue that needs to be addressed, in my opinion, with this widget would be the ability to define the popup window height on line 282.

If it's desired, I could start a PR, but I'm quite unsure about the Qt backend. I would probably heavily orient myself on the last widget added (#3871).

I hope this can help someone out there
[Here's a Slintpad](https://slintpad.com/?snippet=import+%7B+AboutSlint%2C+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++Text+%7B%0A++++++++++++text%3A+%22Hello+World%21%22%3B%0A++++++++++++font-size%3A+24px%3B%0A++++++++++++horizontal-alignment%3A+center%3B%0A++++++++%7D%0A++++++++AboutSlint+%7B%0A++++++++++++preferred-height%3A+150px%3B%0A++++++++%7D%0A++++++++HorizontalLayout+%7B+alignment%3A+center%3B+Button+%7B+text%3A+%22OK%21%22%3B+%7D+%7D%0A++++%7D%0A%7D%0Aimport+%7B+ScrollView%2C+Palette%2C+HorizontalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+FocusBorder+inherits+Rectangle+%7B%0A++++border-width%3A+2px%3B%0A++++border-color%3A+FluentPalette.focus-stroke-outer%3B%0A%0A++++Rectangle+%7B%0A++++++++x%3A+parent.border-width%3B%0A++++++++y%3A+parent.border-width%3B%0A++++++++width%3A+parent.width+-+2+*+parent.border-width%3B%0A++++++++height%3A+parent.height+-+2+*+parent.border-width%3B%0A++++++++border-width%3A+1px%3B%0A++++++++border-radius%3A+parent.border-radius+-+2px%3B%0A++++++++border-color%3A+FluentPalette.focus-stroke-inner%3B%0A++++%7D%0A%7D%0A%0Aexport+component+MenuBorder+inherits+Rectangle+%7B%0A++++border-radius%3A+7px%3B%0A++++background%3A+FluentPalette.alternate-background%3B%0A++++drop-shadow-blur%3A+16px%3B%0A++++drop-shadow-offset-y%3A+8px%3B%0A++++drop-shadow-color%3A+FluentPalette.shadow%3B%0A%0A++++Rectangle+%7B%0A++++++++border-width%3A+1px%3B%0A++++++++border-radius%3A+parent.border-radius%3B%0A++++++++border-color%3A+FluentPalette.control-background-stroke-flyout%3B%0A++++%7D%0A%7D%0A%0Aexport+component+ListItem+%7B%0A++++in+property+%3Cbool%3E+is-selected%3B%0A++++in+property+%3CStandardListViewItem%3E+item%3B%0A++++in+property+%3Cbool%3E+has-focus%3B%0A++++in+property+%3Cbool%3E+has-hover%3B%0A++++in+property+%3Cbool%3E+pressed%3B%0A++++in+property+%3Cint%3E+index%3B%0A++++in+property+%3Clength%3E+pressed-x%3B%0A++++in+property+%3Clength%3E+pressed-y%3B%0A%0A++++min-width%3A+i-layout.min-width%3B%0A++++min-height%3A+max%2840px%2C+i-layout.min-height%29%3B%0A++++vertical-stretch%3A+0%3B%0A++++horizontal-stretch%3A+1%3B%0A%0A++++states+%5B%0A++++++++pressed+when+root.pressed+%3A+%7B%0A++++++++++++i-background.background%3A+is-selected+%3F+%23282828++%3A+%231e1e1e%3B%0A++++++++%7D%0A++++++++hover+when+root.has-hover+%3A+%7B%0A++++++++++++i-text.color%3A+%23d0d0d0%3B%0A++++++++++++i-background.background%3A+is-selected+%3F+%23282828++%3A+%231e1e1e%3B%0A++++++++++++i-selector.height%3A+root.is-selected+%3F+16px+%3A+0%3B%0A++++++++%7D%0A++++++++is-selected+when+root.is-selected+%3A+%7B%0A++++++++++++i-background.background%3A+%232c2c2c%3B%0A++++++++++++i-selector.height%3A+16px%3B%0A++++++++%7D%0A++++%5D%0A%0A++++if+%28root.has-focus%29+%3A+FocusBorder+%7B%0A++++++++border-radius%3A+4px%3B%0A++++%7D%0A%0A++++i-background+%3A%3D+Rectangle+%7B%0A++++++++width%3A+root.width+-+6px%3B%0A++++++++height%3A+root.height+-+4px%3B%0A++++++++background%3A+transparent%3B%0A++++++++border-radius%3A+4px%3B%0A%0A++++++++animate+background+%7B+duration%3A+150ms%3B+%7D%0A%0A++++++++i-layout+%3A%3D+HorizontalLayout+%7B%0A++++++++++++padding-left%3A+16px%3B%0A++++++++++++padding-right%3A+16px%3B%0A++++++++++++spacing%3A+4px%3B%0A%0A++++++++++++i-text+%3A%3D+Text+%7B%0A++++++++++++++++text%3A+root.item.text%3B%0A++++++++++++++++%2F*color%3A+FluentPalette.control-foreground%3B%0A++++++++++++++++font-size%3A+FluentFontSettings.body.font-size%3B%0A++++++++++++++++font-weight%3A+FluentFontSettings.body.font-weight%3B*%2F%0A++++++++++++++++vertical-alignment%3A+center%3B%0A++++++++++++++++horizontal-alignment%3A+left%3B%0A++++++++++++++++overflow%3A+elide%3B%0A%0A++++++++++++++++animate+color+%7B+duration%3A+200ms%3B+%7D%0A++++++++++++%7D%0A++++++++%7D%0A%0A++++++++i-selector+%3A%3D+Rectangle+%7B%0A++++++++++++x%3A+0px%3B%0A++++++++++++y%3A+%28parent.height+-+self.height%29+%2F+2%3B%0A++++++++++++width%3A+3px%3B%0A++++++++++++height%3A+0px%3B%0A++++++++++++background%3A+Palette.accent-background%3B%0A++++++++++++border-radius%3A+2px%3B%0A%0A++++++++++++animate+height+%7B+duration%3A+150ms%3B+easing%3A+ease-out%3B+%7D%0A++++++++%7D%0A++++%7D%0A%0A++++%40children%0A%7D%0Aexport+component+ComboBoxBase+%7B%0A++++in+property+%3C%5Bstring%5D%3E+model%3B%0A++++in+property+%3Cbool%3E+enabled+%3C%3D%3E+i-focus-scope.enabled%3B%0A++++out+property+%3Cbool%3E+has-focus%3A+i-focus-scope.has-focus+%26%26+root.enabled%3B%0A++++out+property+%3Cbool%3E+pressed+%3C%3D%3E+i-touch-area.pressed%3B%0A++++out+property+%3Cbool%3E+has-hover%3A+i-touch-area.has-hover%3B%0A++++in-out+property+%3Cint%3E+current-index%3A+0%3B%0A++++in-out+property+%3Cstring%3E+current-value%3A+root.model%5Broot.current-index%5D%3B%0A%0A++++callback+selected%28%2F*+current-value+*%2F+string%29%3B%0A++++callback+show-popup%28%29%3B%0A%0A++++public+function+select%28index%3A+int%29+%7B%0A++++++++if+%28%21root.enabled%29+%7B%0A++++++++++++return%3B%0A++++++++%7D%0A%0A++++++++root.current-index+%3D+index%3B%0A++++++++root.current-value+%3D+root.model%5Broot.current-index%5D%3B%0A++++++++root.selected%28root.current-value%29%3B%0A++++%7D%0A%0A++++public+function+move-selection-up%28%29+%7B%0A++++++++root.select%28Math.max%28root.current-index+-+1%2C+0%29%29%3B%0A++++%7D%0A%0A++++public+function+move-selection-down%28%29+%7B%0A++++++++root.select%28Math.min%28root.current-index+%2B+1%2C+root.model.length+-+1%29%29%3B%0A++++%7D%0A%0A++++private+property+%3Clength%3E+scroll-delta%3A+2px%3B%0A%0A++++forward-focus%3A+i-focus-scope%3B%0A%0A++++i-focus-scope+%3A%3D+FocusScope+%7B%0A++++++++key-pressed%28event%29+%3D%3E+%7B%0A++++++++++++if+%28%21self.enabled%29+%7B%0A++++++++++++++++return+reject%3B%0A++++++++++++%7D%0A%0A++++++++++++if+%28event.text+%3D%3D+Key.UpArrow%29+%7B%0A++++++++++++++++root.move-selection-up%28%29%3B%0A++++++++++++++++return+accept%3B%0A++++++++++++%7D+else+if+%28event.text+%3D%3D+Key.DownArrow%29+%7B%0A++++++++++++++++root.move-selection-down%28%29%3B%0A++++++++++++++++return+accept%3B%0A++++++++++++%7D+else+if+%28event.text+%3D%3D+Key.Return%29+%7B%0A++++++++++++++++root.show-popup%28%29%3B%0A++++++++++++%7D%0A++++++++++++return+reject%3B%0A++++++++%7D%0A%0A++++++++i-touch-area+%3A%3D+TouchArea+%7B%0A++++++++++++enabled%3A+root.enabled%3B%0A%0A++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++root.focus%28%29%3B%0A++++++++++++++++root.show-popup%28%29%3B%0A++++++++++++%7D%0A%0A++++++++++++scroll-event%28event%29+%3D%3E+%7B%0A++++++++++++++++if+%28event.delta-y+%3C+-root.scroll-delta%29+%7B%0A++++++++++++++++++++root.move-selection-down%28%29%3B%0A++++++++++++++++++++return+accept%3B%0A++++++++++++++++%7D%0A%0A++++++++++++++++if+%28event.delta-y+%3E+root.scroll-delta%29+%7B%0A++++++++++++++++++++root.move-selection-up%28%29%3B%0A++++++++++++++++++++return+accept%3B%0A++++++++++++++++%7D%0A%0A++++++++++++++++reject%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0Aexport+component+ComboBoxScroll+%7B%0A++++in+property+%3C%5Bstring%5D%3E+model+%3C%3D%3E+i-base.model%3B%0A++++in+property+%3Cbool%3E+enabled+%3C%3D%3E+i-base.enabled%3B%0A++++out+property+%3Cbool%3E+has-focus+%3C%3D%3E+i-base.has-focus%3B%0A++++in-out+property+%3Cint%3E+current-index+%3C%3D%3E+i-base.current-index%3B%0A++++in-out+property+%3Cstring%3E+current-value+%3C%3D%3E+i-base.current-value%3B%0A%0A++++callback+selected+%3C%3D%3E+i-base.selected%3B%0A%0A++++min-width%3A+max%28160px%2C+i-layout.min-height%29%3B%0A++++min-height%3A+max%2832px%2C+i-layout.min-height%29%3B%0A++++horizontal-stretch%3A+1%3B%0A++++vertical-stretch%3A+0%3B%0A++++forward-focus%3A+i-base%3B%0A%0A++++states+%5B%0A++++++++disabled+when+%21root.enabled+%3A+%7B%0A++++++++++++i-background.background%3A+FluentPalette.control-disabled%3B%0A++++++++++++i-background.border-color%3A+FluentPalette.border%3B%0A++++++++++++i-text.color%3A+FluentPalette.text-disabled%3B%0A++++++++++++i-icon.colorize%3A+FluentPalette.text-disabled%3B%0A++++++++%7D%0A++++++++pressed+when+i-base.pressed+%3A+%7B%0A++++++++++++i-background.background%3A+FluentPalette.control-alt-tertiary%3B%0A++++++++++++i-background.border-color%3A+FluentPalette.border%3B%0A++++++++++++i-text.color%3A+FluentPalette.text-secondary%3B%0A++++++++++++i-icon.colorize%3A+FluentPalette.text-tertiary%3B%0A++++++++%7D%0A++++++++hover+when+i-base.has-hover+%3A+%7B%0A++++++++++++i-background.background%3A+FluentPalette.control-secondary%3B%0A++++++++%7D%0A++++%5D%0A%0A++++i-base+%3A%3D+ComboBoxBase+%7B%0A++++++++width%3A+100%25%3B%0A++++++++height%3A+100%25%3B%0A%0A++++++++show-popup+%3D%3E+%7B%0A++++++++++++i-popup.show%28%29%3B%0A++++++++%7D%0A++++%7D%0A%0A++++i-background+%3A%3D+Rectangle+%7B%0A++++++++border-radius%3A+3px%3B%0A++++++++background%3A+FluentPalette.non-transperant-control-background%3B%0A++++++++border-width%3A+1px%3B%0A++++++++border-color%3A+FluentPalette.control-border%3B%0A%0A++++++++animate+border-color+%7B+duration%3A+200ms%3B+%7D%0A%0A++++++++i-layout+%3A%3D+HorizontalLayout+%7B%0A++++++++++++padding-left%3A+11px%3B%0A++++++++++++padding-right%3A+11px%3B%0A++++++++++++spacing%3A+8px%3B%0A%0A++++++++++++i-text+%3A%3D+Text+%7B%0A++++++++++++++++horizontal-alignment%3A+left%3B%0A++++++++++++++++vertical-alignment%3A+center%3B%0A++++++++++++++%2F*++font-size%3A+FluentFontSettings.body.font-size%3B%0A++++++++++++++++font-weight%3A+FluentFontSettings.body.font-weight%3B*%2F%0A++++++++++++++++color%3A+FluentPalette.control-foreground%3B%0A++++++++++++++++text%3A+root.current-value%3B%0A++++++++++++%7D%0A%0A++++++++++++i-icon+%3A%3D+Image+%7B%0A++++++++++++++++colorize%3A+FluentPalette.text-secondary%3B%0A++++++++++++++++width%3A+12px%3B%0A++++++++++++++++source%3A+Icons.dropdown%3B%0A++++++++++++++++y%3A+2px%3B%0A%0A++++++++++++++++animate+colorize+%7B+duration%3A+150ms%3B+%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%0A++++%2F%2F+focus+border%0A++++if+%28root.has-focus+%26%26+root.enabled%29+%3A+FocusBorder+%7B%0A++++++++border-radius%3A+i-background.border-radius%3B%0A++++%7D%0A%0A++++i-popup+%3A%3D+PopupWindow+%7B%0A++++++++x%3A+0%3B%0A++++++++y%3A+-4px%3B%0A++++++++width%3A+root.width%3B%0A++++++++height%3A+min%28inner.preferred-width+*+2%29%3B%0A%0A++++++++Rectangle+%7B%0A++++++++++++background%3A+FluentPalette.non-transperant-control-background%3B%0A++++++++++++border-radius%3A+3px%3B%0A++++++++++++border-width%3A+1px%3B%0A++++++++++++border-color%3A+FluentPalette.non-transperant-border%3B%0A++++++++%7D%0A++++++++ScrollView+%7B%0A++++++++++++inner+%3A%3D+VerticalLayout+%7B%0A++++++++++++++++padding%3A+4px%3B%0A++++%0A++++++++++++++++for+value%5Bindex%5D+in+root.model+%3A+ListItem+%7B%0A++++++++++++++++++++item%3A+%7B+text%3A+value+%7D%3B%0A++++++++++++++++++++is-selected%3A+index+%3D%3D+root.current-index%3B%0A++++++++++++++++++++has-hover%3A+i-touch-area.has-hover%3B%0A++++++++++++++++++++pressed%3A+i-touch-area.pressed%3B%0A++++++++++++++++++++i-touch-area+%3A%3D+TouchArea+%7B%0A++++++++++++++++++++++++clicked+%3D%3E+%7B%0A++++++++++++++++++++++++++++i-base.select%28index%29%3B%0A++++++++++++++++++++++++%7D%0A++++++++++++++++++++%7D%0A++++++++++++++++%7D%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D%0A%7D%0A%0A%0A%0A%0A%0A%0A%0A%0A%0A%0A%2F%2F+Copyright+%C2%A9+SixtyFPS+GmbH+%3Cinfo%40slint.dev%3E%0A%2F%2F+SPDX-License-Identifier%3A+GPL-3.0-only+OR+LicenseRef-Slint-Royalty-free-1.2+OR+LicenseRef-Slint-commercial%0Aimport+%7B+Palette+%7D+from+%22std-widgets.slint%22%3B%0A%0A%0Aexport+struct+TextStyle+%7B%0A++++font-size%3A+relative-font-size%2C%0A++++font-weight%3A+int%2C%0A%7D%0A%0Aexport+global+FluentFontSettings+%7B%0A++++out+property+%3Cint%3E+light-font-weight%3A+300%3B%0A++++out+property+%3Cint%3E+regular-font-weight%3A+400%3B%0A++++out+property+%3Cint%3E+semibold-font-weight%3A+600%3B%0A++++out+property+%3CTextStyle%3E+body%3A+%7B%0A++++++++font-size%3A+14+*+0.0769rem%2C%0A++++++++font-weight%3A+regular-font-weight%0A++++%7D%3B%0A++++out+property+%3CTextStyle%3E+body-strong%3A+%7B%0A++++++++font-size%3A+14+*+0.0769rem%2C%0A++++++++font-weight%3A+semibold-font-weight%0A++++%7D%3B%0A%7D%0A%0Aexport+global+FluentPalette+%7B%0A++++in+property+%3Cbool%3E+dark-color-scheme%3A+false%3B%0A%0A++++%2F%2Fnon+transperant+palette%0A++++out+property+%3Cbrush%3E+non-transperant-background%3A+dark-color-scheme++%3F+%231C1C1C+%3A+%23FAFAFA%3B%0A++++out+property+%3Cbrush%3E+non-transperant-foreground%3A+dark-color-scheme++%3F+%23FFFFFF+%3A+%23030303%3B%0A++++out+property+%3Cbrush%3E+non-transperant-control-background%3A+dark-color-scheme+%3F+%232c2c2c+%3A+%23d0d0d0%3B%0A++++out+property+%3Cbrush%3E+non-transperant-control-foreground%3A+dark-color-scheme+%3F+%23FFFFFF+%3A+%23030303%3B%0A++++out+property+%3Cbrush%3E+non-transperant-border%3A+dark-color-scheme+%3F+%23313131+%3A+%23111111%3B%0A%0A++++out+property+%3Cbrush%3E+non-transperant-control-border%3A+dark-color-scheme+%3F+%40linear-gradient%28180deg%2C+%23FFFFFe+0%25%2C+%23ededec+8.33%25%29%0A++++++++%3A+%40linear-gradient%28180deg%2C+%23f0f0ef+90.58%25%2C+%23d6d6d5+100%25%29%3B%0A%0A++++%2F%2F+base+palette%0A++++out+property+%3Cbrush%3E+background%3A+dark-color-scheme+%3F+%231C1C1C+%3A+%23FAFAFA%3B%0A++++out+property+%3Cbrush%3E+foreground%3A+dark-color-scheme+%3F+%23FFFFFF+%3A+%23000000E6%3B%0A++++out+property+%3Cbrush%3E+alternate-background%3A+dark-color-scheme+%3F+%232C2C2C+%3A+%23f0f0f0%3B%0A++++out+property+%3Cbrush%3E+alternate-foreground%3A+dark-color-scheme+%3F+%23FFFFFF+%3A+%23000000E6%3B%0A++++out+property+%3Cbrush%3E+control-background%3A+dark-color-scheme+%3F+%23FFFFFF0F+%3A+%23FFFFFFB3%3B%0A++++out+property+%3Cbrush%3E+control-foreground%3A+dark-color-scheme+%3F+%23FFFFFF+%3A+%23000000E6%3B%0A++++out+property+%3Cbrush%3E+accent-background%3A+dark-color-scheme+%3F+%2360CDFF+%3A+%23005FB8%3B%0A++++out+property+%3Cbrush%3E+accent-foreground%3A+dark-color-scheme+%3F+%23000000+%3A+%23FFFFFF%3B%0A++++out+property+%3Cbrush%3E+selection-background%3A+%230078D4%3B%0A++++out+property+%3Cbrush%3E+selection-foreground%3A+dark-color-scheme+%3F+%23000000+%3A+%23FFFFFF%3B%0A++++out+property+%3Cbrush%3E+border%3A+dark-color-scheme+%3F+%23FFFFFF14+%3A+%2300000073%3B%0A%0A++++%2F%2F+additional+palette%0A++++out+property+%3Cbrush%3E+secondary-accent-background%3A+dark-color-scheme+%3F+%2360CDFFE6+%3A+%23005FB8E6%3B%0A++++out+property+%3Cbrush%3E+tertiary-accent-background%3A+dark-color-scheme+%3F+%2360CDFFCC+%3A+%23005FB8CC%3B%0A++++out+property+%3Cbrush%3E+accent-disabled%3A+dark-color-scheme+%3F+%23FFFFFF29+%3A+%2300000038%3B%0A++++out+property+%3Cbrush%3E+accent-control-border%3A+dark-color-scheme+%3F+%40linear-gradient%28180deg%2C+%23FFFFFF14+90.67%25%2C+%2300000024+100%25%29%0A++++++++%3A+%40linear-gradient%28180deg%2C+%23FFFFFF14+90.67%25%2C+%2300000066+100%25%29%3B%0A++++out+property+%3Cbrush%3E+control-border%3A+dark-color-scheme+%3F+%40linear-gradient%28180deg%2C+%23FFFFFF17+0%25%2C+%2300000012+8.33%25%29%0A++++++++%3A+%40linear-gradient%28180deg%2C+%230000000F+90.58%25%2C+%2300000029+100%25%29%3B%0A++++out+property+%3Cbrush%3E+text-accent-foreground-secondary%3A+dark-color-scheme+%3F+%2300000080+%3A+%23FFFFFFB3%3B%0A++++out+property+%3Cbrush%3E+text-accent-foreground-disabled%3A+dark-color-scheme+%3F+%23FFFFFF87+%3A+%23FFFFFF%3B%0A++++out+property+%3Cbrush%3E+text-secondary%3A+dark-color-scheme+%3F+%23FFFFFFC9+%3A+%2300000099%3B%0A++++out+property+%3Cbrush%3E+text-tertiary%3A+dark-color-scheme+%3F+%23FFFFFF8A+%3A+%2300000073%3B%0A++++out+property+%3Cbrush%3E+text-disabled%3A+dark-color-scheme+%3F+%23FFFFFF5E+%3A+%230000005E%3B%0A++++out+property+%3Cbrush%3E+text-control-border%3A+dark-color-scheme+%3F+%40linear-gradient%28180deg%2C+%23FFFFFF14+99.98%25%2C+%23FFFFFF8A+100%25%2C+%23FFFFFF8A+100%25%29%0A++++++++%3A+%40linear-gradient%28180deg%2C+%230000000F+99.99%25%2C+%2300000073+100%25%2C+%2300000073+100%25%29%3B%0A++++out+property+%3Cbrush%3E+control-secondary%3A+dark-color-scheme+%3F+%23FFFFFF14+%3A+%23F9F9F980%3B%0A++++out+property+%3Cbrush%3E+control-tertiary%3A+dark-color-scheme+%3F+%23FFFFFF08+%3A+%23F9F9F94D%3B%0A++++out+property+%3Cbrush%3E+control-disabled%3A+dark-color-scheme+%3F+%23FFFFFF0A+%3A+%23F9F9F94D%3B%0A++++out+property+%3Cbrush%3E+control-alt-secondary%3A+dark-color-scheme+%3F+%230000001A+%3A+%2300000005%3B%0A++++out+property+%3Cbrush%3E+control-alt-tertiary%3A+dark-color-scheme+%3F+%23FFFFFF0A+%3A+%230000000F%3B%0A++++out+property+%3Cbrush%3E+control-alt-quartiary%3A+dark-color-scheme+%3F+%23FFFFFF12+%3A+%2300000017%3B%0A++++out+property+%3Cbrush%3E+control-alt-disabled%3A+transparent%3B%0A++++out+property+%3Cbrush%3E+control-strong-stroke%3A+dark-color-scheme+%3F+%23FFFFFF99+%3A+%2300000099%3B%0A++++out+property+%3Cbrush%3E+control-strong-stroke-disabled%3A+dark-color-scheme+%3F+%23FFFFFF29+%3A+%2300000038%3B%0A++++out+property+%3Cbrush%3E+control-solid%3A+dark-color-scheme+%3F+%23454545%3A+%23FFFFFF%3B%0A++++out+property+%3Cbrush%3E+circle-border%3A+dark-color-scheme+%3F+%40linear-gradient%28180deg%2C+%23FFFFFF17+0%25%2C+%23FFFFFF12+100%25%29%0A++++%3A+%40linear-gradient%28180deg%2C+%230000000F+0%25%2C+%2300000029+100%25%29%3B%0A++++out+property+%3Cbrush%3E+control-input-active%3A+dark-color-scheme+%3F+%231E1E1EB3+%3A+%23FFFFFF%3B%0A++++out+property+%3Cbrush%3E+focus-stroke-inner%3A+dark-color-scheme+%3F+%23000000B3+%3A+%23FFFFFF%3B%0A++++out+property+%3Cbrush%3E+focus-stroke-outer%3A+dark-color-scheme+%3F+%23FFFFFF+%3A+%23000000E6%3B%0A++++out+property+%3Cbrush%3E+control-background-stroke-flyout%3A+dark-color-scheme+%3F+%2300000033+%3A+%230000000F%3B%0A++++out+property+%3Cbrush%3E+sub-title-secondary%3A+dark-color-scheme+%3F+%23FFFFFF0F+%3A+%230000000A%3B%0A++++out+property+%3Cbrush%3E+sub-title-tertiary%3A+dark-color-scheme+%3F+%23FFFFFF0A+%3A+%2300000005%3B%0A++++out+property+%3Cbrush%3E+shadow%3A+dark-color-scheme+%3F+%2300000042+%3A+%2300000024%3B%0A++++out+property+%3Cbrush%3E+subtle%3A+dark-color-scheme+%3F+%23FFFFFF0F+%3A+%230000000A%3B%0A++++out+property+%3Cbrush%3E+subtle-secondary%3A+dark-color-scheme+%3F+%23FFFFFF0F+%3A+%230000000A%3B%0A++++out+property+%3Cbrush%3E+subtle-tertiary%3A+dark-color-scheme+%3F+%23FFFFFF0A+%3A+%2300000005%3B%0A++++out+property+%3Cbrush%3E+divider%3A+dark-color-scheme+%3F+%23FFFFFF14+%3A+%2300000014%3B%0A++++out+property+%3Cbrush%3E+layer-on-mica-base-alt%3A+dark-color-scheme+%3F+%233A3A3A73+%3A+%23FFFFFFB3%3B%0A++++out+property+%3Cbrush%3E+layer-on-mica-base-alt-secondary%3A+dark-color-scheme+%3F+%23FFFFFF0F+%3A+%230000000A%3B%0A++++out+property+%3Cbrush%3E+card-stroke%3A+dark-color-scheme+%3F+%230000001A+%3A+%230000000F%3B%0A%7D%0A%0Aexport+global+Icons+%7B%0A++++out+property+%3Cimage%3E+arrow-down%3A+%40image-url%28%22_arrow-down.svg%22%29%3B%0A++++out+property+%3Cimage%3E+arrow-up%3A+%40image-url%28%22_arrow-up.svg%22%29%3B%0A++++out+property+%3Cimage%3E+check-mark%3A+%40image-url%28%22_check-mark.svg%22%29%3B%0A++++out+property+%3Cimage%3E+chevron-down%3A+%40image-url%28%22_chevron-down.svg%22%29%3B%0A++++out+property+%3Cimage%3E+chevron-up%3A+%40image-url%28%22_chevron-up.svg%22%29%3B%0A++++out+property+%3Cimage%3E+down%3A+%40image-url%28%22_down.svg%22%29%3B%0A++++out+property+%3Cimage%3E+dropdown%3A+%40image-url%28%22_dropdown.svg%22%29%3B%0A++++out+property+%3Cimage%3E+left%3A+%40image-url%28%22_left.svg%22%29%3B%0A++++out+property+%3Cimage%3E+right%3A+%40image-url%28%22_right.svg%22%29%3B%0A++++out+property+%3Cimage%3E+up%3A+%40image-url%28%22_up.svg%22%29%3B%0A%7D%0A%0A%0A%0A%0A%0A%0A%0A%0A%0A%0A%0Aimport+%7B+Palette%2C+ComboBox%2C+HorizontalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+MainWindow+inherits+Window+%7B%0A++++height%3A+300px%3B%0A++++title%3A+%40tr%28%22Reelix%22%29%3B%0A++++HorizontalBox%7B%0A++++ComboBoxScroll+%7B%0A++++height%3A+40px%3B%0A++++width%3A+240px%3B%0A++++model%3A+%5B%22Item1%22%2C%22Item2%22%2C%22Item3%22%2C%22Item4%22%2C%22Item5%22%2C%22Item6%22%2C%22Item7%22%2C%22Item8%22%2C%22Item9%22%2C%22Item10%22%5D%3B%0A++++%7D%0A++++ComboBox+%7B%0A++++height%3A+40px%3B%0A++++width%3A+240px%3B%0A++++model%3A+%5B%22Item1%22%2C%22Item2%22%2C%22Item3%22%2C%22Item4%22%2C%22Item5%22%2C%22Item6%22%2C%22Item7%22%2C%22Item8%22%2C%22Item9%22%2C%22Item10%22%5D%3B%0A++++%7D%0A++++%7D%0A++++++++init+%3D%3E+%7B%0A++++if+%28Palette.color-scheme+%3D%3D+ColorScheme.unknown%29+%7B%0A++++++++++++FluentPalette.dark-color-scheme+%3D+false%3B%0A++++%7D%0A++++if+%28Palette.color-scheme+%3D%3D+ColorScheme.light%29+%7B%0A++++++++++++FluentPalette.dark-color-scheme+%3D+false%3B%0A++++%7D%0A++++if+%28Palette.color-scheme+%3D%3D+ColorScheme.dark%29+%7B%0A++++++++++++FluentPalette.dark-color-scheme+%3D+true%3B%0A++++%7D%0A++++%7D%0A%7D&style=native)(The Icon is not showing up because you can't define a icon in Slintpad and the light theme is a bit broken)
![Animation](https://github.com/slint-ui/slint/assets/96253562/574568aa-0555-4491-b7ee-8747c7712dd0)

**Comment by ghost** [comment, marked as answer]
Hi, thank you very much for your proposal. Looks great. I have also thought about to implement it similar. The optimal solution would be that the popup stretches to a defined max height and then start to scroll if not all items fit in the max height. I tried it but with currently it is difficult to solve with `PopupWindow`. Until it is fixed for the popup I think the solution with a fixed popup window height and scroll could be a good solution.
